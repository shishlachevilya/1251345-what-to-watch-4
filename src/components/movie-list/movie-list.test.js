import React from "react";
import renderer from "react-test-renderer";
import MovieList from "./movie-list";
import {BrowserRouter as Router} from "react-router-dom";

const movies = [
  {
    id: `1`,
    path: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`
  },
  {
    id: `2`,
    path: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`
  },
  {
    id: `3`,
    path: `img/macbeth.jpg`,
    title: `Macbeth`
  },
  {
    id: `4`,
    path: `img/aviator.jpg`,
    title: `Aviator`
  },
  {
    id: `5`,
    path: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`
  },
  {
    id: `6`,
    path: `img/what-we-do-in-the-shadows.jpg`,
    title: `What We Do in the Shadows`
  },
  {
    id: `7`,
    path: `img/revenant.jpg`,
    title: `Revenant`
  },
  {
    id: `8`,
    path: `img/johnny-english.jpg`,
    title: `Johnny English`
  },
];

describe(`MovieList component test`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
    .create(
        <Router>
          <MovieList
            movies={movies}
            onCardTitleClick={() => {}}
          />
        </Router>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
