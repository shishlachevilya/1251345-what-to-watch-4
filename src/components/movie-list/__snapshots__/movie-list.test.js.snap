// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component test renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    id="1"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="2"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="3"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="4"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="5"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="6"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="7"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="8"
    onMouseOver={[Function]}
  >
    <a
      href="/detail"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <img
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="/detail"
        onClick={[Function]}
      />
    </h3>
  </article>
</div>
`;
